ci/docker/debug-cpu-stdexec.yaml

```diff
@@ -18,9 +18,18 @@ spack:
       true
 
   specs:
-    - dla-future@master +miniapps +ci-test +ci-check-threads ^intel-mkl threads=openmp ^mpich ^pika+stdexec ^stdexec@git.48c52df0f81c6151eecf4f39fa5eed2dc0216204=main
+    - dla-future@master build_type=Debug +miniapps +ci-test +ci-check-threads
```

Comment:
In `common.yaml`, `threads=openmp` is only a variant of `intel-mkl`. My current understanding is that variants have low precedence. Do we need
```
intel-mkl:
  require:
  - "threads=openmp"
```
to ensure the same behavior?